superset-frontend/src/explore/components/controls/DatasourceControl/index.jsx

```diff
@@ -264,7 +302,11 @@ class DatasourceControl extends React.PureComponent {
             <WarningIconWithTooltip warningMarkdown={extra.warning_markdown} />
           )}
           <AntdDropdown
-            overlay={datasourceMenu}
+            overlay={
+              isValidDatasourceType(datasource.type)
+                ? queryDatasourceMenu
```

Comment:
is it just showing the query now for testing?

Reply:
This only shows the query dropdown if the datasource is a query, it should not be showing at this point.